lib/clients/api-client.js

```diff
@@ -70,8 +70,12 @@ SlackAPIClient.prototype.send = function(method, options, callback) {
       if (Array.isArray(options.attachments)) {
         options.attachments = JSON.stringify(options.attachments);
       }
-      requestOptions.method = 'GET';
-      requestOptions.qs = options;
+
+      // We default to a POST request here because long messages will exceed the ~2000 character
+      // limit in the url for a GET request.
+      requestOptions.method = 'POST';
+      requestOptions.headers = {'Content-Type': 'application/x-www-form-urlencoded'};
+      requestOptions.body = querystring.stringify(options);
```

Comment:
@FredKSchott Yes. This StackOverflow answer hints that they may not support JSON. http://stackoverflow.com/a/33310661/1180335. Which I was kind of surprised about, and couldn't find anything in the docs about it.  I tried passing `requestOptions.body = JSON.stringify(options)` with the appropriate header, but kept getting a `not authed` error back. Switched to the above and it worked so I stopped exploring the JSON format. 

I've tested on my branch and error I was seeing went away/messages were posted successfully.